### ISSUE TITLE
Corebird fails to play videos/gifs

### ISSUE THREAD

**Issue by Flat** [baedert/corebird#744]
using master branch:
```
gstvideodecoder.c(1156): gst_video_decoder_sink_event_default (): /GstPlayBin:video/GstURIDecodeBin:uridecodebin0/GstDecodeBin:decodebin0/avdec_h264:avdec_h264-0:
no valid frames found
```

**Comment by baedert** [comment]
Works fine here.

**Comment by Vistaus** [comment]
Also working fine here. Are you sure you have installed the required gstreamer packages and their respective dev(el) packages?

**Comment by Flat** [comment]
```
 sudo pacman -Qs gst-
local/gst-libav 1.12.1-1
    GStreamer Multimedia Framework ffmpeg Plugin
local/gst-plugins-bad 1.12.1-1
    GStreamer Multimedia Framework Bad Plugins
local/gst-plugins-base 1.12.1-1
    GStreamer Multimedia Framework Base Plugins
local/gst-plugins-base-libs 1.12.1-1
    GStreamer Multimedia Framework Base Plugin libraries
local/gst-plugins-good 1.12.1-1
    GStreamer Multimedia Framework Good Plugins
local/gst-plugins-ugly 1.12.1-1
    GStreamer Multimedia Framework Ugly Plugins
```

Nvidia 650Ti Boost - Arch Linux

**Comment by Vistaus** [comment]
Hmm, well I'm currently on gstreamer 1.10.4 (that's the newest Solus provides currently) and like I said: it's working fine for me. So maybe it's a bug in 1.12.1?

**Comment by IBBoard** [comment]
I (might) get the same issue on my openSUSE Tumbleweed machine but not on my openSUSE Leap machine. I'll see if I can diagnose it when I'm next on my Tumbleweed machine.

**Comment by Flat** [comment]
```
local/gst-libav 1.10.4-1
    GStreamer Multimedia Framework ffmpeg Plugin
local/gst-plugins-bad 1.10.4-7
    GStreamer Multimedia Framework Bad Plugins
local/gst-plugins-base 1.10.4-1
    GStreamer Multimedia Framework Base Plugins
local/gst-plugins-base-libs 1.10.4-1
    GStreamer Multimedia Framework Base Plugin libraries
local/gst-plugins-good 1.10.4-1
    GStreamer Multimedia Framework Good Plugins
local/gst-plugins-ugly 1.10.4-1
    GStreamer Multimedia Framework Ugly Plugins
local/gstreamer 1.10.4-1
    GStreamer Multimedia Framework
local/gstreamer-vaapi 1.10.4-1
    GStreamer Multimedia Framework VAAPI Plugin
```
Downgraded to 1.10.4 and recompiled corebird. Same error.

**Comment by diddledan** [comment]
Does your GStreamer installation include the h264 codec such as one backed by the libopenh264 library by Cisco? I recommend openh264 if you can use the binary from their site because you can be sure you're properly licensed to use the patent-encumbered codec. If you do not want to use the binary release from Cisco and be bound by its patent licensing, then you can compile the library yourself or use a package provided by your distro if they've packaged it.

The error message in your first post indicated that GStreamer can't find any h264 plugin, so you will be unable to play videos until you can figure out which toggles to wiggle to get GStreamer supporting the h264 codec. It might be that GStreamer doesn't know where the correct library is, or you might not have a valid library installed yet.

**Comment by Flat** [comment]
```
gst-inspect-1.0 | grep h264
rtp:  rtph264depay: RTP H264 depayloader
rtp:  rtph264pay: RTP H264 payloader
uvch264:  uvch264mjpgdemux: UVC H264 MJPG Demuxer
uvch264:  uvch264src: UVC H264 Source
videoparsersbad:  h264parse: H.264 parser
libav:  avdec_h264: libav H.264 / AVC / MPEG-4 AVC / MPEG-4 part 10 decoder
typefindfunctions: video/x-h264: h264, x264, 26
```
`gst-play-1.0` also has no problem playing h264 encoded video.

**Comment by Vistaus** [comment]
Strange. Is there anyone who can confirm Flat's issue?

**Comment by IBBoard** [comment]
I've tried on my other machine and I can't recreate it now - videos and what I think are GIFs both run now (on openSUSE Tumbleweed). Something must have updated, or it was the rebuild to Master.

**Comment by Flat** [comment]
Confirmed it is a gstreamer issue - not sure what the exact issue is yet. Will comment if I ever figure it out. Thanks for the assistance in debugging; sorry it wasn't a problem with corebird

**Comment by Flat** [comment]
If anyone else comes across this problem it turned out to be using ffmpeg from git master. Using the ffmpeg provided by the distro fixed the issue. Either mismatch between gstreamer and ffmpeg or something that's changed in master.

**Comment by ViniciusBRodrigues** [comment]
I'm receiving this error:

```
gsturidecodebin.c(1006): no_more_pads_full (): /GstPlayBin:video/GstURIDecodeBin:uridecodebin0:
no suitable plugins found:
gstdecodebin2.c(4565): gst_decode_bin_expose (): /GstPlayBin:video/GstURIDecodeBin:uridecodebin0/GstDecodeBin:decodebin0:
no suitable plugins found:
Missing decoder: MPEG-4 AAC (audio/mpeg, framed=(boolean)true, mpegversion=(int)4, level=(string)2, base-profile=(string)lc, profile=(string)lc, rate=(int)44100, channels=(int)1, stream-format=(string)adts)
Missing decoder: H.264 (video/x-h264, stream-format=(string)byte-stream, alignment=(string)nal, parsed=(boolean)true)
```

**Comment by baedert** [comment]
You'll need to install the gstreamer package that provides the h.264 decoder. What package that is, I don't know. It's usually something like `gst-libav`. Or fix the package in your distro. ffs.

**Comment by ViniciusBRodrigues** [comment]
I already have installed _libx264-148_, which is "a free h264/avc encoder - encoder binary"

**Comment by baedert** [comment]
But can gstreamer use it? Why does the opensuse package not depend no the appropriate gstreamer package?

**Comment by ViniciusBRodrigues** [comment]
I don't know :(
This is what I get when I search for h264. I also have the gstreamer plugins reallated to h264 acoordting to this site: https://gstreamer.freedesktop.org/documentation/plugins.html, which are gst-plugins-ugly and bad.

**Comment by baedert** [comment]
Try installing `gstreamer-plugins-libav`. If that still doesn't work, ask on some opensuse help forum/channel, I don't have knowledge over every package in every distribution.

**Comment by IBBoard** [comment]
I hit this recently after replacing Leap with Tumbleweed and re-installing all of my apps.

The default openSUSE package doesn't pull in quite the right dependencies, but [my (Git Master plus tweaks) version should do](https://build.opensuse.org/package/view_file/home:IBBoard:desktop/corebird/corebird.spec?expand=1). The problem appears to be that `gstreamer-plugins-bad` claims to support h264, but that isn't enough for Twitter videos. You need `gstreamer-plugins-libav` for Twitter videos.

From a bit of investigation then I believe it specifically needs something that provides `gstreamer1(decoder-video/x-compressed-yuv)()(64bit)` (because ffprobe reports h264 YUV420p and libav provides that but "bad" doesn't).

**Comment by ViniciusBRodrigues** [comment]
Hi, thank you - `gstreamer-plugins-libav` worked fine.

**Comment by IBBoard** [comment]
I just tripped over the original issue (`decodebin0/avdec_h264:avdec_h264-0: no valid frames found`, not the "no suitable plugins" follow-up) in the latest openSUSE Tumbleweed.

If I run `gst-play-1.0 --gst-debug-level=1 https://video.twimg.com/tweet_video/DN47pBXXkAI4bU0.mp4` then I currently get a stack of `ERROR libav :0:: Got unexpected packet size after a partial decode`.

Searching around, I found [this Arch Linux bug](https://bugs.archlinux.org/task/56065) from a couple of weeks ago that references ffmpeg 3.4 vs 3.3.4. It also references [this GStreamer bug](https://bugzilla.gnome.org/show_bug.cgi?id=789193), which references [this upstream ffmpeg regression](https://trac.ffmpeg.org/ticket/6775).

So, not Corebird's problem in either case, but a ffmpeg issue in the first case and missing plugins in the second case.

(Adding the information in case it helps people find it or understand it).
